File: Exercises/JavaScript/JavaScript_Basics/Variable_Scope/ex6.js
// What will the following code log to the console and why? Don't run it until you have tried to answer.

let a = 5;
let b = false;

if (a > 4) {
  let b = true;
}

console.log(b);

// it will print false because 'b' is re-scoped by the let statement inside the if block, so outside the if block 'b' will remain the same value

File: Exercises/Python/Python_Basics/6_Variable-Scope/ex6.py
'''
What will the following code do and why? Don't run it until you have tried to answer.
'''

a = 1

def my_function():
    a = 2

my_function()
print(a)

'''
it will print 1, the function creates a local variable 'a' that does not affect the scope above it
'''
